src/main/java/bio/terra/workspace/service/trace/StackdriverTrace.java

```diff
@@ -0,0 +1,63 @@
+package bio.terra.workspace.service.trace;
+
+import bio.terra.workspace.app.configuration.StackdriverConfiguration;
+import bio.terra.workspace.common.exception.StackdriverRegistrationException;
+import bio.terra.workspace.common.utils.GoogleUtils;
+import io.opencensus.common.Scope;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceConfiguration;
+import io.opencensus.exporter.trace.stackdriver.StackdriverTraceExporter;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.samplers.Samplers;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StackdriverTrace {
+  private final StackdriverConfiguration stackdriverConfiguration;
+
+
+  @Autowired
+  public StackdriverTrace(StackdriverConfiguration stackdriverConfiguration) {
+    this.stackdriverConfiguration = stackdriverConfiguration;
+    // Creates and registers the exporter. This must be happen each time WSM is instantiated.
+    // createAndRegister();
```

Comment:
When I call createAndRegister() here, I get a bunch of unit test failures of this type:

`DataReferenceServiceTest > enumerateFailsUnauthorized() STARTED

DataReferenceServiceTest > enumerateFailsUnauthorized() FAILED
    java.lang.IllegalStateException at DefaultCacheAwareContextLoaderDelegate.java:132
        Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException at ConstructorResolver.java:797
            Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException at ConstructorResolver.java:797
                Caused by: org.springframework.beans.factory.BeanCreationException at ConstructorResolver.java:313
                    Caused by: org.springframework.beans.BeanInstantiationException at BeanUtils.java:217
                        Caused by: java.lang.IllegalStateException at Preconditions.java:511
`

Reply:
I imagine it's having some issues starting the exporter for tests? The quick fix for this would be to start the exporter in StartUpInitializer. But, I don't know if you have some java magic that will help here :)

Reply:
`org.springframework.beans.factory.UnsatisfiedDependencyException` means some component isn't getting all of its dependencies wired in correctly - when I ran it, I was getting errors pointing to StackdriverConfiguration not being able to find the SA credentials in tests, but that was because I hadn't rendered the file

Does this work when called from StartUpInitializer?